### ISSUE TITLE
Repeated type on key input on Android device using a Bluetooth keyboard

### ISSUE THREAD

**Issue by aryoxp** [WebCoder49/code-input#124]
When I type using a Bluetooth keyboard on an Android device, every time I press a character on keyboard, which I had previously typed before, is reappearing on code-input editor area.

In the following screenshot on the demo page CodePen, on line 8 of the editor, I press the 'a' key, then 'b', 'c', 'd', and 'e'.
The expected output on the editor should be '`abcde`', however, I got '`aababcabcdabcde`'. The previously typed letter reappear on the editor.
 
![Screenshot_20241220_135849_Chrome](https://github.com/user-attachments/assets/1d49a1dd-96b1-4906-b59f-4f9c22f72090)

This is not happening when I used the on-screen keyboard of the tablet device.
My device is Samsung Galaxy Tab S8, Android 14, Google Chrome web browser.
The Bluetooth keyboard I was using is Logitech Keys-To-Go-2; 
the same device, browser, and keyboard that I used to type this report.
I don't have other Bluetooth external keyboard, so I don't know whether the problem is on the keyboard incompatibility or a bug in the code-input key detection/sync.

Any suggestion to this issue is appreciated. Thank you.

**Comment by WebCoder49** [comment]
Thanks for the detailed description, Aryo. I think this might be similar to bug #117. Please could you let me know what behaviour you get when you do the same with [this demo](https://codepen.io/WebCoder49/pen/vYMpMoJ?editors=1100)?

**Comment by aryoxp** [comment]
Yes, it is the same with [the demo](https://codepen.io/WebCoder49/pen/vYMpMoJ?editors=1100).

![Screenshot_20241220_174933_Chrome](https://github.com/user-attachments/assets/908ba77e-f7da-42d6-9670-488f4b3c09c4)

**Comment by WebCoder49** [comment]
Thanks for the quick reply. I will let you know when I have changes ready for you to test/actions to test.

**Comment by WebCoder49** [comment]
I just realised I sent you the wrong demo, @aryoxp. Please could you try out [this one (without the indent plugin which may be affecting the behaviour)](https://codepen.io/WebCoder49/pen/GgKExMV) and let me know what behaviour you get?

Also, it would be helpful if you could try out both this demo and the previous one on all browsers you have on the Android tablet as well as [Mozilla Firefox](https://play.google.com/store/apps/details?id=org.mozilla.firefox), since it uses a different rendering engine to Chrome so may act differently. Thanks!

**Comment by aryoxp** [comment]
sorry for the late reply,
in Chrome, the behaviour is still the same.
![Screenshot_20241229_165105_Chrome](https://github.com/user-attachments/assets/f26602ff-1d47-4155-b5c6-a5735d80464f)

in Firefox, the behaviour is even weirder, some characters only appear once, but duplicating the rest uncontrollably.
![Screenshot_20241229_165817_Firefox](https://github.com/user-attachments/assets/45e48f69-c109-4749-a934-29c601486546)

in the above screenshot, I typed: 'abcde', 'qwert', 'asdfg', and 'zxcvb', one on each line., however, it seems that the [Enter] key is not repeating.

**Comment by WebCoder49** [comment]
For your information, I'm very busy right now until 21 June. I can't implement new feature requests or bugfixes myself in projects I maintain, but can merge contributions by others, especially if they fix bugs. I've therefore unassigned this from myself for now.

**Comment by WebCoder49** [comment]
I can verify the above message is legitimate.

**Comment by WebCoder49** [comment]
@aryoxp I should have notified you above. Please could you test as described above?

**Comment by WebCoder49** [comment]
I believe this is fixed after #149, and can be tested with the code in the `main` branch right now or in the next release.
